src/messages.rs

```diff
@@ -363,3 +364,169 @@ pub enum ResponseContent {
+            RequestContent::Post(ref data, ref message_id) =>
+                write!(formatter, "RequestContent {{ Post( {:?}, {:?} ) }}", data, message_id),
+            RequestContent::Delete(ref data, ref message_id) =>
+                write!(formatter, "RequestContent {{ Delete( {:?}, {:?} ) }}", data, message_id),
+        }
+    }
+}
+
+impl fmt::Debug for ResponseContent {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match *self {
+            ResponseContent::GetNetworkName { ref relocated_id } =>
+                write!(formatter, "ResponseContent {{ GetNetworkName {{ {:?} }} }}", relocated_id),
+            ResponseContent::GetPublicId { ref public_id } =>
+                write!(formatter, "ResponseContent {{ GetPublicId {{ {:?} }} }}", public_id),
+            ResponseContent::GetPublicIdWithEndpoints { ref public_id, .. } =>
+                write!(formatter, "ResponseContent {{ GetPublicIdWithEndpoints {{ {:?}, .. }} }}", public_id),
+            ResponseContent::GetCloseGroup { ref close_group_ids } =>
+                write!(formatter, "ResponseContent {{ GetCloseGroup {{ {:?} }} }}", close_group_ids),
+            ResponseContent::GetSuccess(ref data, ref message_id) =>
+                write!(formatter, "ResponseContent {{ GetSuccess {{ {:?}, {:?} }} }}", data, message_id),
+            ResponseContent::PutSuccess(ref digest, ref message_id) =>
+                write!(formatter, "ResponseContent {{ PutSuccess {{ {:?}, {:?} }} }}", digest, message_id),
+            ResponseContent::PostSuccess(ref digest, ref message_id) =>
+                write!(formatter, "ResponseContent {{ PostSuccess {{ {:?}, {:?} }} }}", digest, message_id),
+            ResponseContent::DeleteSuccess(ref digest, ref message_id) =>
+                write!(formatter, "ResponseContent {{ DeleteSuccess {{ {:?}, {:?} }} }}", digest, message_id),
+            ResponseContent::GetFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ GetFailure {{ {:?}, {:?}, .. }} }}", id, request),
+            ResponseContent::PutFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ PutFailure {{ {:?}, {:?}, .. }} }}", id, request),
+            ResponseContent::PostFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ PostFailure {{ {:?}, {:?}, .. }} }}", id, request),
+            ResponseContent::DeleteFailure { ref id, ref request, .. } =>
+                write!(formatter, "ResponseContent {{ DeleteFailure {{ {:?}, {:?}, .. }} }}", id, request),
+        }
+    }
+}
+
+
+#[cfg(test)]
+mod test {
+    extern crate rand;
+
+    use super::{HopMessage, SignedMessage, RoutingMessage, RequestMessage, RequestContent};
+    use id::FullId;
+    use authority::Authority;
+    use xor_name::XorName;
+    use sodiumoxide::crypto::sign;
+
+    #[test]
+    fn signed_message_check_integrity() {
```

Comment:
Looks like we're not really checking anything but the success cases here.  Could we add checks for failure of integrity if any of the three fields of the signed message is tampered with or corrupted please?